### PR TITLE
Jamesweb dev

### PR DESCRIPTION
**Pull request by jamesweb-ms**
Bug fixes

### CODE REVIEW

File: Common/Deployment/DeploymentLib.ps1 [hunk 1]
```diff
@@ -234,20 +253,13 @@ function GetAzureStorageAccount()
         [Parameter(Mandatory=$true,Position=0)] [string] $storageBaseName,
         [Parameter(Mandatory=$true,Position=1)] [string] $resourceGroupName
     )
-    $storageAccountName = ValidateResourceName $storageBaseName.ToLowerInvariant().Replace('-','') Microsoft.Storage/storageAccounts $resourceGroupName
+    $storgeTempName = $storageBaseName.ToLowerInvariant().Replace('-','')
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Is this a typo? (`$storgeTempName`)

File: Common/Deployment/DeploymentLib.ps1 [hunk 2]
```diff
@@ -234,20 +253,13 @@ function GetAzureStorageAccount()
...
+    $storageAccountName = ValidateResourceName $storgeTempName.Substring(0, [System.Math]::Min(24, $storgeTempName.Length)) Microsoft.Storage/storageAccounts $resourceGroupName
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: And here too in the `Min(24, $storgeTempName`...?

File: Common/Deployment/DeploymentLib.ps1 [hunk 3]
```diff
@@ -546,7 +565,11 @@ function GetAADTenant()
     if ($result.value.Count -eq 0)
     {
         $body = ReplaceFileParameters ("{0}\Application.json" -f $global:azurePath) -arguments @($global:site, $global:environmentName)
-        $result = Invoke-RestMethod -Method "POST" -Uri $uri -Headers @{"Authorization"=$header;"Content-Type"="application/json"} -Body $body -ErrorAction Stop
+        $result = Invoke-RestMethod -Method "POST" -Uri $uri -Headers @{"Authorization"=$header;"Content-Type"="application/json"} -Body $body -ErrorAction SilentlyContinue
+        if ($result -eq $null)
+        {
+            throw "Unable create application'$($global:site)iotsuite'"
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Minor grammar, but should this be `Unable to create application`?

File: Common/Deployment/DeploymentLib.ps1 [hunk 4]
```diff
@@ -563,7 +586,11 @@ function GetAADTenant()
     if ($result.value.Count -eq 0)
     {
         $body = "{ `"appId`": `"$applicationId`" }"
-        $result = Invoke-RestMethod -Method "POST" -Uri $uri -Headers @{"Authorization"=$header;"Content-Type"="application/json"} -Body $body -ErrorAction Stop
+        $result = Invoke-RestMethod -Method "POST" -Uri $uri -Headers @{"Authorization"=$header;"Content-Type"="application/json"} -Body $body -ErrorAction SilentlyContinue
+        if ($result -eq $null)
+        {
+            throw "Unable create ServicePrincipal for application '$($global:site)iotsuite'"
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Minor grammar again--missing `to`.

File: Common/Deployment/DeploymentLib.ps1 [hunk 5]
```diff
@@ -581,8 +608,15 @@ function GetAADTenant()
     if (($result.value | ?{$_.ResourceId -eq $resourceId}) -eq $null)
     {
         $body = "{ `"id`": `"$roleId`", `"principalId`": `"$($authResult.UserInfo.UniqueId)`", `"resourceId`": `"$resourceId`" }"
-        $result = Invoke-RestMethod -Method "POST" -Uri $uri -Headers @{"Authorization"=$header;"Content-Type"="application/json"} -Body $body -ErrorAction Stop
-        Write-Host "Successfully assigned user to application '$($result.resourceDisplayName)' as role 'Admin'"
+        $result = Invoke-RestMethod -Method "POST" -Uri $uri -Headers @{"Authorization"=$header;"Content-Type"="application/json"} -Body $body -ErrorAction SilentlyContinue
+        if ($result -eq $null)
+        {
+            Write-Warning "Unable create RoleAssignment for application '$($global:site)iotsuite' for current user - will be Implicit Readonly"
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Missing `to`?

File: Common/Deployment/DeploymentLib.ps1 [hunk 6]
```diff
@@ -190,6 +208,7 @@ function ValidateResourceName()
         "microsoft.storage/storageaccounts"
         {
             $resourceUrl = "blob.core.windows.net"
+            $resourceBaseName = $resourceBaseName.Substring(0, [System.Math]::Min(19, $storgeTempName.Length))
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Variable typo at end of line

Comment (jamesweb-ms) [inline Common/Deployment/DeploymentLib.ps1]: Yah, not even right... should be $resourceBaseName

**Pull request merged**